openmp/libomptarget/plugins-nextgen/amdgpu/src/rtl.cpp

```diff
@@ -1848,6 +1848,7 @@ struct AMDGPUDeviceTy : public GenericDeviceTy, AMDGenericDeviceTy {
         OMPX_StreamBusyWait("LIBOMPTARGET_AMDGPU_STREAM_BUSYWAIT", 2000000),
         OMPX_UseMultipleSdmaEngines(
             "LIBOMPTARGET_AMDGPU_USE_MULTIPLE_SDMA_ENGINES", false),
+        HSAXnackEnv("HSA_XNACK", false), OMPX_ApuMaps("OMPX_APU_MAPS", false),
```

Comment:
I really don't think we should be parsing HSA's environment variables redundantly. This information is available from the system target in the form of `amdgcn-amd-amdhsa-gfx...:xnack+` right? We should use that instead.

Reply:
Which ROCr function returns that string for the system (and not for the binary)?

Reply:
hsa_isa_get_info_alt with HSA_ISA_INFO_NAME. I went through this code recently w/ Saiyedul and it should respect the environment variable AFAIK.